pyk/src/pyk/rpc/rpc.py

```diff
@@ -71,37 +72,105 @@ class JsonRpcMethod(Protocol):
     def __call__(self, **kwargs: Any) -> Any: ...
 
 
-class JsonRpcRequestHandler(BaseHTTPRequestHandler):
-    methods: dict[str, JsonRpcMethod]
+@dataclass(frozen=True)
+class JsonRpcRequest:
```

Comment:
Consider just making this a `typing.NamedTuple`:

```suggestion
class JsonRpcRequest(NamedTuple):
```